### PR TITLE
Convert ReadService to suspend functions

### PR DESCRIPTION
**Pull request by Dominaezzz**
Signed-off-by: Dominic Fischer <dominicfischer7@gmail.com>

### Pull Request Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [x] Pull request is based on the develop branch
- [ ] Pull request updates [CHANGES.md](https://github.com/vector-im/element-android/blob/develop/CHANGES.md)
- [x] Pull request includes screenshots or videos if containing UI changes
- [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailViewModel.kt [hunk 1]
```diff
@@ -181,7 +181,9 @@ class RoomDetailViewModel @AssistedInject constructor(
         observePowerLevel()
         updateShowDialerOptionState()
         room.getRoomSummaryLive()
-        room.markAsRead(ReadService.MarkAsReadParams.READ_RECEIPT, NoOpMatrixCallback())
+        viewModelScope.launch {
+            room.markAsRead(ReadService.MarkAsReadParams.READ_RECEIPT)
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailViewModel.kt:185]: The several call should be protected by try catch block I think

**Review by bmarty — COMMENTED** [review]
Thanks
One question

File: vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailViewModel.kt (hunk 1 above)
Comment (Dominaezzz) [reply vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailViewModel.kt:185]: Ah you're right. I'll add that.

**Review by bmarty — APPROVED** [review]
Thanks for the update!

**Pull request merged**